services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/attribute/LocalDateAttributeConverter.java

```diff
@@ -24,16 +24,17 @@
 import software.amazon.awssdk.enhanced.dynamodb.AttributeConverter;
 import software.amazon.awssdk.enhanced.dynamodb.AttributeValueType;
 import software.amazon.awssdk.enhanced.dynamodb.EnhancedType;
+import software.amazon.awssdk.enhanced.dynamodb.internal.converter.ConverterUtils;
 import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
 
 /**
  * A converter between {@link LocalDate} and {@link AttributeValue}.
  *
  * <p>
- * This stores and reads values in DynamoDB as a number, so that they can be sorted numerically as part of a sort key.
+ * This stores and reads values in DynamoDB as a String.
  *
  * <p>
- * LocalDateTimes are stored in the format "[-]YYYYMMDD000000", where:
+ * LocalDateTimes are stored in the format "[-]YYYY-MM-DDT00:00:00Z", where:
```

Comment:
LocalDate should not store timezone information. Writing a valid ISO8601 output here risks it being read as an Instant and breaking the data-loss rule. I would propose having it write "YYYY-MM-DD" and allowing LocalDateTime to parse that format.